timemachine/cpp/src/barostat.cu

```diff
@@ -287,34 +142,31 @@ void MonteCarloBarostat::inplace_move(
         return;
     }
 
-    curandErrchk(curandSetStream(cr_rng_, stream));
-    // Generate scaling and metropolis conditions in one pass
-    curandErrchk(curandGenerateUniformDouble(cr_rng_, d_rand_, 2));
-
-    gpuErrchk(cudaMemsetAsync(d_init_u_, 0, sizeof(*d_init_u_), stream));
-    gpuErrchk(cudaMemsetAsync(d_final_u_, 0, sizeof(*d_final_u_), stream));
-
-    gpuErrchk(cudaMemsetAsync(d_u_buffer_, 0, bps_.size() * sizeof(*d_u_buffer_), stream));
-    gpuErrchk(cudaMemsetAsync(d_u_after_buffer_, 0, bps_.size() * sizeof(*d_u_after_buffer_), stream));
-
-    runner_.execute_potentials(bps_, N_, d_x, d_box, nullptr, nullptr, d_u_buffer_, stream);
-    accumulate_energy(bps_.size(), d_u_buffer_, d_init_u_, stream);
+    // Get offset into the d_rand_ array
+    int random_offset = (((step_ / interval_) * 2) - 2) % (RANDOM_BATCH_SIZE * 2);
 
-    k_setup_barostat_move<<<1, 1, 0, stream>>>(d_rand_, d_box, d_volume_delta_, d_volume_scale_, d_length_scale_);
-    gpuErrchk(cudaPeekAtLastError());
+    // Generate the scaling and metropolis conditions in batches then offset on each move
```

Comment:
nit: comment unclear to me - is this generating two random numbers, one to be used for the box scaling and the other to be used for generating the accept/reject decision?

Reply:
Attempted to clarify in [5e40597](https://github.com/proteneer/timemachine/pull/1101/commits/5e4059737c2f59e9d2b5a670aa69e152cc186c3c)